ledger-core/application/testwalletapi/statemachine/statemachine_test.go

```diff
@@ -118,43 +124,72 @@ func TestSMTestAPICall_Migrate_After_SendRequest(t *testing.T) {
 		requestPayload: request,
 	}
 
-	smWrapper := slotMachine.AddStateMachine(ctx, &smRequest)
-
-	slotMachine.RunTil(smWrapper.AfterStep(smRequest.stepRegisterBargeIn))
+	p1 := slotMachine.PulseSlot.CurrentPulseNumber()
 
-	outgoingRef := smRequest.requestPayload.CallOutgoing
-	_, bargeIn := slotMachine.SlotMachine.GetPublishedGlobalAliasAndBargeIn(outgoingRef)
-	require.NotNil(t, bargeIn)
+	smWrapper := slotMachine.AddStateMachine(ctx, &smRequest)
 
 	messageSent := make(chan struct{}, 1)
 	slotMachine.MessageSender.SendRole.Set(func(_ context.Context, msg payload.Marshaler, role node.DynamicRole, object reference.Global, pn pulse.Number, _ ...messagesender.SendOption) error {
 		res := msg.(*payload.VCallRequest)
+
 		// ensure that both times request is the same
-		assert.Equal(t, outgoingRef, res.CallOutgoing)
+		assert.Equal(t, APICaller, res.Caller)
+		assert.Equal(t, APICaller.GetBase(), res.CallOutgoing.GetBase())
+		assert.Equal(t, p1, res.CallOutgoing.GetLocal().GetPulseNumber())
 		assert.Equal(t, node.DynamicRoleVirtualExecutor, role)
-		assert.Equal(t, request.Callee, object)
+		assert.Equal(t, reference.NewSelf(res.CallOutgoing.GetLocal()), object)
 
 		messageSent <- struct{}{}
 		return nil
 	})
 
-	slotMachine.RunTil(smWrapper.AfterStep(smRequest.stepSendRequest))
+	slotMachine.RunTil(smWrapper.BeforeStep(smRequest.stepProcessResult))
 	testutils.WaitSignalsTimed(t, 10*time.Second, messageSent)
-	slotMachine.Migrate()
+}
 
-	slotMachine.RunTil(smWrapper.AfterStep(smRequest.stepSendRequest))
-	testutils.WaitSignalsTimed(t, 10*time.Second, messageSent)
+func TestSMTestAPICall_RetriesExceeded(t *testing.T) {
+	var (
+		mc  = minimock.NewController(t)
+		ctx = instestlogger.TestContext(t)
+	)
 
-	response := &payload.VCallResult{
-		Caller:   gen.UniqueGlobalRef(),
-		Callee:   gen.UniqueGlobalRef(),
-		CallAsOf: gen.PulseNumber(),
+	slotMachine := slotdebugger.New(ctx, t)
+
+	request := payload.VCallRequest{
+		CallType:       payload.CTMethod,
+		Callee:         gen.UniqueGlobalRef(),
+		CallFlags:      payload.BuildCallFlags(contract.CallTolerable, contract.CallDirty),
+		CallSiteMethod: "New",
+		Arguments:      []byte("some args"),
 	}
 
-	// simulate received VCallResult
-	require.True(t, bargeIn.CallWithParam(response))
+	slotMachine.PrepareMockedMessageSender(mc)
+
+	slotMachine.Start()
+	defer slotMachine.Stop()
+
+	smRequest := SMTestAPICall{
+		requestPayload: request,
+	}
+
+	smWrapper := slotMachine.AddStateMachine(ctx, &smRequest)
+
+	messageSent := make(chan struct{}, 1)
+	slotMachine.MessageSender.SendRole.Set(func(_ context.Context, msg payload.Marshaler, role node.DynamicRole, object reference.Global, pn pulse.Number, _ ...messagesender.SendOption) error {
+		messageSent <- struct{}{}
+		return nil
+	})
+
+	for i := 0; i < 3; i++ {
```

Comment:
may be replace '3' -> 'MaxRepeats' contant

Reply:
nope, this is a test, change of number of max retries in code should result in test failure, I will add a constant

Reply:
added a constant

Reply:
hmmmm. okay!